### PR TITLE
test(eda.create_report): added test script for create_report function

### LINKED ISSUE #236: The documentation and tests is missing for create_report
TODO:
- [ ] Add documentation for create report
- [ ] Fix the docstring for create_report (mode is not documented)
- [ ] Add tests for create report (codecov is down)

### PR DESCRIPTION
**Pull request by eutialia**
# Description

This commit also adds new documentation for create_report function.

Resolve #236

# How Has This Been Tested?

Test scripts

# Snapshots:

N/A

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have already squashed the commits and make the commit message conform to the project standard.
- [x] I have already marked the commit with "BREAKING CHANGE" or "Fixes #" if needed.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: dataprep/eda/create_report/formatter.py [hunk 1]
```diff
@@ -27,6 +27,20 @@
...
+        Currently only the 'basic' is fully implemented.
+
+    Returns
+    -------
+    Dict[str, Any]
+        A dictionary in which formatted data will be stored.
+        This variable acts like an API in passing data to the template engine.
     """
 
     comps: Dict[str, Any] = {}
```
Comment (brandonlockhart) [inline dataprep/eda/create_report/formatter.py:46]: Why is an empty dict passed to each `format_` method?

File: docs/source/eda/create_report.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+==================================================================
+
+.. toctree::
+   :maxdepth: 2
+
+Overview
+========
+
+The goal of `create_report` is to generate profile reports from a pandas DataFrame. `create_report` utilizes the functionalities and formats the plots from `dataprep`. It provides the following information:
+
```
[the remark this answers is not in the document]
Comment (eutialia) [reply docs/source/eda/create_report.rst:13]: Pandas DataFrame is the only type a user will use in `create_report`

File: dataprep/eda/create_report/formatter.py (hunk 1 above)
Comment (eutialia) [reply dataprep/eda/create_report/formatter.py:46]: `format_` functions will add raw data to this dictionary. Those functions act like a pipeline to the input and return expected output.

File: docs/source/eda/create_report.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    df = pd.read_csv("https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv")
+
+Generate report via `create_report(df)`
+=======================================
+
+After getting a dataset, we could generate the report by calling `create_report(df)`. The following shows an example::
+
+    from dataprep.eda import create_report
+    create_report(df)
+
```
Comment (dovahcrow) [inline docs/source/eda/create_report.rst:38]: I think maybe it is better to provide a link for the reader to download the generated report before we introduce each section.

**Pull request merged**